covalent/cloud_resource_manager/core.py

```diff
@@ -0,0 +1,242 @@
+
+        Returns:
+            Exit code of the process
+        """
+        proc = subprocess.Popen(
+            args=cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            cwd=workdir,
+            shell=True,
+            env=env_vars,
+        )
+        retcode = self._print_stdout(proc)
+
+        if retcode != 0:
+            raise subprocess.CalledProcessError(returncode=retcode, cmd=cmd)
+
+    def _update_config(self, tf_executor_config_file: str) -> None:
+        """
+        Update covalent configuration with the executor
+        config values as obtained from terraform
+        """
+
+        # Puts the plugin options in covalent's config
+        executor_config = ConfigParser()
+        executor_config.read(tf_executor_config_file)
+        for key, value in executor_config[self.executor_name].items():
+            converted_value = get_converted_value(value)
+            set_config({f"executors.{self.executor_name}.{key}": converted_value})
+
+    def _get_tf_path(self) -> str:
+        """
+        Get the terraform path
+        """
+        if terraform := shutil.which("terraform"):
+            return terraform
+        else:
+            raise CommandNotFoundError("Terraform not found on system")
+
+    def up(self, dry_run: bool = True):
+        """
+        Setup executor resources
+        """
+        terraform = self._get_tf_path()
+
+        tfvars_file = str(Path(self.executor_tf_path) / "terraform.tfvars")
+        tf_executor_config_file = str(Path(self.executor_tf_path) / f"{self.executor_name}.conf")
+
+        tf_init = " ".join([terraform, "init"])
+        tf_plan = " ".join([terraform, "plan", "-out", "tf.plan"])
+        tf_apply = " ".join([terraform, "apply", "tf.plan"])
+
+        # Run `terraform init`
+        self._run_in_subprocess(cmd=tf_init, workdir=self.executor_tf_path)
+
+        # Setup terraform infra variables as passed by the user
+        tf_vars_env_dict = os.environ.copy()
+        if self.executor_options:
+            with open(tfvars_file, "w") as f:
+                for key, value in self.executor_options.items():
+                    tf_vars_env_dict[f"TF_VAR_{key}"] = value
+
+                    # Write whatever the user has passed to the terraform.tfvars file
+                    f.write(f'{key}="{value}"\n')
+
+        # Run `terraform plan`
+        cmd_output = self._run_in_subprocess(
+            cmd=tf_plan, workdir=self.executor_tf_path, env_vars=tf_vars_env_dict
+        )
+
+        # Create infrastructure as per the plan
+        # Run `terraform apply`
+        if not dry_run:
+            cmd_output = self._run_in_subprocess(
+                cmd=tf_apply, workdir=self.executor_tf_path, env_vars=tf_vars_env_dict
+            )
+
+            # Update covalent executor config based on Terraform output
+            self._update_config(tf_executor_config_file)
+
+        return cmd_output
```

Comment:
(non blocking question) What kind of info do we expect in the `cmd_output` and is it useful for anything?